tests/runtimes/simple-keyvalue/src/lib.rs

```diff
@@ -0,0 +1,270 @@
+            Ok(cbor::to_value(&Self::get(ctx, args)?))
+        }();
+        match result {
+            Ok(value) => CallResult::Ok(value),
+            Err(err) => err.to_call_result(),
+        }
+    }
+
+    fn _callable_remove_handler(
+        _mi: &CallableMethodInfo,
+        ctx: &mut TxContext,
+        body: cbor::Value,
+    ) -> CallResult {
+        let result = || -> Result<cbor::Value, Error> {
+            let args = cbor::from_value(body).map_err(|_| Error::InvalidArgument)?;
+            Ok(cbor::to_value(&Self::remove(ctx, args)?))
+        }();
+        match result {
+            Ok(value) => CallResult::Ok(value),
+            Err(err) => err.to_call_result(),
+        }
+    }
+}
+
+// Actual implementation of this runtime's externally-callable methods.
+impl Module {
+    // Insert given keyvalue into storage.
+    fn insert(ctx: &mut TxContext, body: KeyValue) -> Result<(), Error> {
+        if ctx.is_check_only() {
+            return Ok(());
+        }
+        let mut store = sdk::storage::PrefixStore::new(ctx.runtime_state(), &MODULE_NAME);
+        let mut ts = sdk::storage::TypedStore::new(&mut store);
+        ts.insert(body.key, &body.value);
+        Ok(())
+    }
+
+    // Fetch keyvalue from storage using given key.
+    fn get(ctx: &mut TxContext, body: Key) -> Result<KeyValue, Error> {
+        if ctx.is_check_only() {
+            return Ok(KeyValue {
+                key: body.key,
+                value: vec![],
+            });
+        }
+        let mut store = sdk::storage::PrefixStore::new(ctx.runtime_state(), &MODULE_NAME);
+        let ts = sdk::storage::TypedStore::new(&mut store);
+        let v: Vec<u8> = ts.get(body.key.clone()).ok_or(Error::InvalidArgument)?;
+        Ok(KeyValue {
+            key: body.key,
+            value: v,
+        })
+    }
+
+    // Remove keyvalue from storage using given key.
+    fn remove(ctx: &mut TxContext, body: Key) -> Result<(), Error> {
+        if ctx.is_check_only() {
+            return Ok(());
+        }
+        let mut store = sdk::storage::PrefixStore::new(ctx.runtime_state(), &MODULE_NAME);
+        let mut ts = sdk::storage::TypedStore::new(&mut store);
+        ts.remove(body.key);
+        Ok(())
+    }
+}
+
+impl sdk::module::MigrationHandler for Module {
+    type Genesis = Genesis;
+
+    fn init_or_migrate(
+        ctx: &mut DispatchContext,
+        meta: &mut sdk::modules::core::types::Metadata,
+        genesis: &Self::Genesis,
+    ) -> bool {
+        let version = meta.versions.get(Self::NAME).copied().unwrap_or_default();
+        if version == 0 {
+            // Initialize state from genesis.
+            Self::set_params(ctx.runtime_state(), &genesis.parameters);
+            meta.versions.insert(Self::NAME.to_owned(), Self::VERSION);
+            return true;
+        }
+
+        // Migrations are not supported.
+        false
+    }
+}
+
+/// Simple keyvalue runtime.
+pub struct Runtime;
```

Comment:
Would be nice if the `Module` was separate from the runtime (doesn't need to be its own crate just in its own Rust module).